nilearn/decomposition/canica.py

```diff
@@ -106,52 +107,42 @@ class CanICA(MultiPCA, CacheMixin):
       datasets", IEEE ISBI 2010, p. 1177
     """
 
-    def __init__(self, mask=None, n_components=20,
-                 smoothing_fwhm=6, do_cca=True,
-                 threshold='auto', n_init=10,
-                 standardize=True,
-                 random_state=0,
-                 target_affine=None, target_shape=None,
+    def __init__(self, n_init=10, threshold='auto', n_components=20,
+                 do_cca=True,
+                 random_state=None,
+                 mask=None, smoothing_fwhm=None,
+                 standardize=True, detrend=True,
```

Comment:
Why did you reorder the arguments to CanICA? It's potentially a backward incompatible change, and I don't think that we want it.